addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Device.java

```diff
@@ -0,0 +1,352 @@
+/**
+ * Copyright (c) 2010-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.km200.internal;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The KM200Device representing the device with its all capabilities
+ *
+ * @author Markus Eckhardt
+ *
+ */
+
+public class KM200Device {
+
+    private static final Logger logger = LoggerFactory.getLogger(KM200Device.class);
+
+    /* valid IPv4 address of the KMxxx. */
+    protected String ip4Address = null;
+
+    /* The gateway password which is provided on the type sign of the KMxxx. */
+    protected String gatewayPassword = null;
+
+    /* The private password which has been defined by the user via EasyControl. */
+    protected String privatePassword = null;
+
+    /* The returned device charset for communication */
+    protected String charSet = null;
+
+    /* Needed keys for the communication */
+    protected byte[] cryptKeyInit = null;
+    protected byte[] cryptKeyPriv = null;
+
+    /* Buderus_MD5Salt */
+    protected byte[] MD5Salt = null;
+
+    /* Device services */
+    public HashMap<String, KM200CommObject> serviceTreeMap = null;
+
+    /* Device services blacklist */
+    List<String> blacklistMap = null;
+    /* List of virtual services */
+    List<KM200CommObject> virtualList = null;
+
+    /* Is the first INIT done */
+    protected Boolean inited = false;
+
+    public KM200Device() {
+        serviceTreeMap = new HashMap<String, KM200CommObject>();
+        blacklistMap = new ArrayList<String>();
+        blacklistMap.add("/gateway/firmware");
+        virtualList = new ArrayList<KM200CommObject>();
+    }
+
+    public Boolean isConfigured() {
+        if (StringUtils.isNotBlank(ip4Address) && cryptKeyPriv != null) {
```

Comment:
this may be shortened to `return StringUtils.isNotBlank(ip4Address) && cryptKeyPriv != null;`